documentation/projects/proposals/additional_search_views/20230719-implementation_plan_additional_search_views.md

```diff
@@ -0,0 +1,366 @@
+quoted value.
+[https://api.openverse.engineering/v1/images/?tags=%22black%20cat%22](https://api.openverse.engineering/v1/images/?source=met)
+This query returns all images with tags that contain the phrase "black cat".
+
+This would not match the tags exactly, but tags page is supposed to show all
+items that are associated with the same topic or category.
+
+We could also add a media view at `/api/v1/<media>/tags/?q=<tag_name>`, but it
+is difficult to implement because the tags are stored as a list JSONField model.
+
+### Nuxt store changes
+
+We can re-use the search store as is for these pages. Currently, the store uses
+the `searchBy` filter to determine the shape of the API query. If `searchBy`
+value is set, then the `q` parameter is replaced with the
+`<searchBy>=<searchTerm>` query parameter. API query parameters are constructed
+from the search store state in the
+[`prepare-search-query-params` method](https://github.com/WordPress/openverse/blob/b4b46b903731870015c475d2c08eebef7ec6b25b/frontend/src/utils/prepare-search-query-params.ts#L22-L25)
+
+We can update the `searchBy` filter to have one of the possible values:
+`{ searchBy: <null|creator|source|tag> }`. Then, this value would be used to
+construct the API query.
+
+#### Update the `searchBy` filter
+
+The `searchBy` filter will be used to determine the shape of the API query.
+While currently these parameters will be mutually exclusive (we can only search
+by one of them), we might want to allow searching by multiple parameters in the
+future.
+
+For other filters, we only use `toggle` method to update the value. However, for
+`searchBy`, we need to be able to check one of the `searchBy` parameters, and
+uncheck the others. To enable that, we should add a new `search` store method.
+
+This change should also update `prepare-search-query-params` to use the
+`searchBy` filter value to construct the API query:
+
+```typescript
+const prepareSearchQuery = (
+  searchParams: Record<string, string>,
+  mediaType: SupportedMediaType
+) => {
+  if (searchParams.searchBy) {
+    if (searchBy === "creator") {
+      // the name of the source is stored in the `<mediaType>Provider` filter
+      return {
+        creator: searchTerm,
+        source: searchParams[`${mediaType}Provider`],
+      }
+    } else if (searchBy === "source") {
+      return { source: searchParams[`${mediaType}Provider`] }
+    } else if (searchBy === "tag") {
+      // The parameter is plural in the API
+      return { tags: searchTerm }
+    }
+  } else {
+    // Current search query transform
+  }
+}
+```
+
+### Page changes
+
+#### Create a page for `tag` / `creator` /`source` collections.
+
+Nuxt allows creating nested dynamic routes like
+`/pages/_collection/_mediaType/_term`.
+
+Here, the collection can be `tag`, `creator` or `source`. `mediaType` can be one
+of the supported media types (currently, `image` and `audio`). `term` refers to
+the actual value of the tag, creator or source name.
```

Comment:
Realising now that for these to work, creator needs to have source included. It could make sense to make these match the API routes: `/_mediaType/source/_source/creator/_creator`, `/_mediaType/tag/_tag` etc.